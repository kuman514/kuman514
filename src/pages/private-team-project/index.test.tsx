import { render } from '@testing-library/react';

import PrivateTeamProjectPage from '.';

describe('PrivateTeamProjectPage', () => {
  it('should show as expected', () => {
    const { container } = render(<PrivateTeamProjectPage />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
