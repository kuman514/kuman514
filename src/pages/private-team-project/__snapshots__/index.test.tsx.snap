// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateTeamProjectPage should show as expected 1`] = `
<main
  class="flex min-h-screen flex-col px-8 py-24 gap-4 sm:px-24"
>
  <h1
    class="text-bold text-4xl sm:text-5xl md:text-6xl"
  >
    비공개 팀 프로젝트
  </h1>
  <h2
    class="text-bold text-4xl"
  >
    사용한 기술 스택
  </h2>
  <div
    class="flex flex-row flex-wrap gap-2"
  >
    <span
      class="font-mono bg-gray-300 text-red-700 text-sm px-1 py-0.5 rounded undefined"
    >
      Expo
    </span>
    <span
      class="font-mono bg-gray-300 text-red-700 text-sm px-1 py-0.5 rounded undefined"
    >
      React Native
    </span>
  </div>
  <h2
    class="text-bold text-4xl"
  >
    개요
  </h2>
  <ul
    class="list-disc pl-8"
  >
    <li>
      팀 구성: PM 1명, 기획자 1명, 디자이너 1명, 백엔드 2명, 프론트엔드 1명 (본인)
    </li>
    <li>
      구현 내용
      <ul
        class="list-disc pl-8"
      >
        <li>
          Expo로 React Native 프로젝트를 세팅하여, iOS + Android 타겟이면서도 MacOS뿐만 아니라 Windows로도 개발할 수 있게 만듦.
        </li>
        <li>
          React Navigation의 Native Stack Navigator를 사용하여 스크린 라우팅.
        </li>
        <li>
          Axios를 이용하여 게시물 등록/수정/삭제/열람과 이미지 업로드 등등, 백엔드 API와 연계된 각종 기능 구현.
        </li>
        <li>
          Amazon Cognito Identity JS를 이용하여 회원가입, 로그인, 회원탈퇴, 비밀번호 초기화, 이메일 인증 등의 유저 관련 기능 구현.
        </li>
        <li>
          각종 기능에 대해 로딩 여부, 에러 여부, 결과값을 담은 커스텀 훅을 구축.
        </li>
        <li>
          expo-updates 패키지와 Expo Channel을 활용한 CI/CD를 구축하여, Expo 무료 플랜에서 월 30회(iOS 15회)까지만 할 수 있는 EAS Build를 절약함.
        </li>
      </ul>
    </li>
  </ul>
</main>
`;
